### ISSUE TITLE
Improve accessibility of page elements

### ISSUE THREAD

**Issue by NSoiffer** [benetech/MathShare#52]
It is very difficult to navigate a page via AT without structure, which is the current situation.

Here's a plan on how to fix accessibility:
- Header: on the div, we should add role="heading" aria-level="1"
  - On the "mathHistory" div, add role="heading" aria-level="2"
    - On the step lablels (step 1, etc), add role="heading" aria-level="3". Also create an offscreen span that says "1 step" so something is read for screen readers. Number should be first. Ideally it would be ordinal numbers such "first step", but "1 step" better. Don't make users have to listen to "step" first.
      - On the math in each step, add role="heading" aria-level="4"
      - On the annotation in each step, add role="heading" aria-level="4"
  - On the math current area, add role="heading" aria-level="2"
    - on the math input area, change div to section and add role="heading" aria-level="3" along with aria-label="type math here"
    - on the speech input area, add role="heading" aria-level="3"
    - on the palette, add role="heading" aria-level="3". Change the div to section and add aria-label="math input buttons"

On the palette buttons, don't use "title". Find some other jquery method for titles. Create an invisible span at the end of each button content that says the keyboard equivalent.

**Comment by sueannma** [comment]
someone pointed us to [AViewer](https://developer.paciellogroup.com/resources/aviewer/) (Windows accessibility checker) to highlight things that might make the user experience for screen reader users better. I tried really hard to understand the details, but am sure I missed something (and some might not be relevant now since it was tested on a Linux machine using Firefox): 
* overuse of headers
* lack of labeling (names) for headers
* tool tips usage (consider using title/description and other related tags, names also should be words rather than symbols)
* input fields should use more appropriate ARIA tags (and not tagged as headers)
* think some buttons (on the home page?) might've been missing proper names

**Comment by sueannma** [comment]
navigation overview/instructions available on the github wiki

**Comment by afijal** [comment]
So I've started with fixing errors that tests from https://github.com/benetech/MathShare/issues/84 give us.
Should I change font/bacground colors to resolve such issues: 

> Element has insufficient color contrast of 3.03 (foreground color: #ffffff, background color: #5997e2, font size: 17.3pt, font weight: normal). Expected contrast ratio of 4.5:1

**Comment by afijal** [comment]
for example, currently the only violation is contrast of this button:
![image](https://user-images.githubusercontent.com/2682379/41715707-9a8021ce-7554-11e8-9d2c-5fb690a40861.png)
I'm no expert on accessibility but it looks contrasty enough to me

**Comment by afijal** [comment]
@johnhbenetech 
So we will leave contrast related issues for now as stated during the meeting, here is the rest of questions:
-that invisible span, should it really be a keyboard equivalent? I guess what currently sits in the button's title would fit here, e.g. "Square root (⌨: sqrt)"
-do we need title in buttons at all?
-finally, would you recommend any tool/plugin that would work on linux to check the accessibility or can we rely solely on tests from #84?

**Comment by afijal** [comment]
@johnhbenetech I've used that AViewer, but I couldn't find any automatic audit option that would return us a list of things that are wrong or should be improved, is there such a functionality or rather it's a tool for manual checking?

**Comment by johnhbenetech** [comment]
Yeah, it's used for manual review. It could be used as a second opinion when we are explicitly working on a11y improvements.

**Comment by afijal** [comment]
@johnhbenetech  So I will use it to conduct a manual review, as discussed on the meeting we will stick to wcag level AA, right?
Also @NSoiffer, John told that you may suggest to use some additional tools except AViewer, any suggestions?

**Comment by afijal** [comment]
@johnhbenetech 
Ok so here are the results:
https://docs.google.com/spreadsheets/d/1o7fQAZWkDEvsgwMRXszkZgKY0kc_pxUyIodvxQGMdpQ/edit?usp=sharing

Some notes/questions:
-help is currently aria-hidden - why? Should it stay like that?
-we don't care about contrast atm but even at a first glance, edit button looks like it should be more contrasty
-history steps seem aria hidden too, also, should they be accessible by keyboard?
